xbmc/platform/darwin/osx/XBMCApplication.mm

```diff
@@ -134,15 +116,10 @@ @implementation XBMCDelegate
 // Set the working directory to the .app's parent directory
 - (void)setupWorkingDirectory
 {
-  char parentdir[MAXPATHLEN];
-  CFURLRef url = CFBundleCopyBundleURL(CFBundleGetMainBundle());
-  CFURLRef url2 = CFURLCreateCopyDeletingLastPathComponent(0, url);
-  if (CFURLGetFileSystemRepresentation(url2, true, (UInt8*)parentdir, MAXPATHLEN))
-  {
-    assert(chdir(parentdir) == 0); /* chdir to the binary app's parent */
-  }
-  CFRelease(url);
-  CFRelease(url2);
+  NSString* bundlePath = [[NSBundle mainBundle] resourcePath];
+  NSString* mainPath = [bundlePath stringByDeletingLastPathComponent];
+
+  assert(chdir(mainPath.UTF8String) == 0); /* chdir to the binary app's parent */
```

Comment:
If Kodi is in /Applications/Kodi.app, original code would point to /Applications
```suggestion
  auto parentPath = NSBundle.mainBundle.bundlePath.stringByDeletingLastPathComponent;
  NSAssert([NSFileManager.defaultManager changeCurrentDirectoryPath:parentPath]);
```

I see that this code was copied from the old implementation, but which purpose it serves? Is it still required?

Reply:
RE the required state, Id honestly prefer to just leave it as is for now. Dont want to go chasing shit when theres already so much other stuff happening in this PR. Ive no idea what its used for, or why its there, So i'll add a todo to check if its still relevant at a later point in time

Reply:
ok, let's keep it, just apply the suggestion please